File: android/designsystem/src/main/kotlin/com/bolsillo/designsystem/theme/Confidence.kt
package com.bolsillo.designsystem.theme

import androidx.compose.runtime.Immutable
import androidx.compose.ui.graphics.Color

// Confidence buckets mirror shared-assets/design/tokens.json → confidence.
// The threshold default 0.75 lives in domain (ExpenseClassifier.DEFAULT_THRESHOLD,
// Article V); this module accepts it as a parameter so :designsystem stays
// dependency-free of :domain.

enum class ConfidenceState { Waiting, Low, High }

@Immutable
data class ConfidenceVisual(
    val state: ConfidenceState,
    val foreground: Color,
    val container: Color,
    val chipBorder: Color?,
    val labelKey: String,
)

fun confidenceVisual(
    confidence: Double,
    threshold: Double,
    colors: BolsilloColors,
): ConfidenceVisual =
    when {
        confidence <= 0.0 ->
            ConfidenceVisual(
                state = ConfidenceState.Waiting,
                foreground = colors.textMuted,
                container = colors.fill,
                chipBorder = null,
                labelKey = "record.ai.waiting",
            )
        confidence < threshold ->
            ConfidenceVisual(
                state = ConfidenceState.Low,
                foreground = colors.warning,
                container = colors.warningContainer,
                chipBorder = colors.warningBorder,
                labelKey = "record.ai.toConfirm",
            )
        else ->
            ConfidenceVisual(
                state = ConfidenceState.High,
                foreground = colors.success,
                container = colors.successContainer,
                chipBorder = null,
                labelKey = "record.ai.confident",
            )
    }

File: ios/Packages/Bolsillo/Sources/BolsilloDesignSystem/Theme/Confidence.swift
import SwiftUI

public enum ConfidenceState {
    case waiting
    case low
    case confident
}

public struct ConfidenceVisual {
    public let icon: String
    public let fg: KeyPath<BolsilloColors, Color>
    public let container: KeyPath<BolsilloColors, Color>
    public let labelKey: String
    public let chipBorder: KeyPath<BolsilloColors, Color>?

    public init(
        icon: String,
        fg: KeyPath<BolsilloColors, Color>,
        container: KeyPath<BolsilloColors, Color>,
        labelKey: String,
        chipBorder: KeyPath<BolsilloColors, Color>? = nil
    ) {
        self.icon       = icon
        self.fg         = fg
        self.container  = container
        self.labelKey   = labelKey
        self.chipBorder = chipBorder
    }
}

public func confidenceState(_ confidence: Double, threshold: Double = 0.75) -> ConfidenceState {
    if confidence <= 0 { return .waiting }
    if confidence < threshold { return .low }
    return .confident
}

public func confidenceVisual(for state: ConfidenceState) -> ConfidenceVisual {
    switch state {
    case .waiting:
        return ConfidenceVisual(
            icon:      "sparkles",
            fg:        \.textMuted,
            container: \.fill,
            labelKey:  "record.ai.waiting"
        )
    case .low:
        return ConfidenceVisual(
            icon:      "info.circle",
            fg:        \.warning,
            container: \.warningContainer,
            labelKey:  "record.ai.toConfirm",
            chipBorder: \.warningBorder
        )
    case .confident:
        return ConfidenceVisual(
            icon:      "checkmark.circle.fill",
            fg:        \.success,
            container: \.successContainer,
            labelKey:  "record.ai.confident"
        )
    }
}
